### PR TITLE
+ Labels to DSS Parameters

### PR DESCRIPTION
**Pull request by Marsobad**
- I added a "Label" attribute to DSSParameters to have more flexibility on the error messages.
- I fixed a few unit tests.

### CODE REVIEW

**Review by StanislasGuinel — APPROVED** [review]
Nice addition :)

File: dkulib/dku_config/dss_parameter.py [hunk 1]
```diff
@@ -21,8 +21,9 @@ class DSSParameter:
         required(bool, optional): Whether the value can be None
         cast_to(type, optional): The type to cast the variable in
         cast_to(Any, optional): The default value of the variable (If value is None)
+        label(str, optional): The name displayed to the end user
     """
-    def __init__(self, name: str, value: Any, checks: List[dict] = None, required: bool = False, cast_to: type = None, default: Any = None):
+    def __init__(self, name: str, value: Any, checks: List[dict] = None, required: bool = False, cast_to: type = None, default: Any = None, label: str = None,):
```
Comment (HenriChabert) [inline dkulib/dku_config/dss_parameter.py:26]: I like to keep a logical order to the arguments (This is very subjective, you can change it if you don't agree):
```suggestion
    def __init__(self, name: str, value: Any, label: str = None, checks: List[dict] = None, required: bool = False, default: Any = None, cast_to: type = None):
```

If doing so, please also change the order of the documentation

File: dkulib/dku_config/dss_parameter.py [hunk 2]
```diff
@@ -40,6 +42,10 @@ def __init__(self, name: str, value: Any, checks: List[dict] = None, required: b
         self.required = required
         self.cast_to = cast_to
         self.checks = [CustomCheck(**check) for check in checks]
+        if label is None:
+            self.label = name
+        else:
+            self.label = label
```
Comment (HenriChabert) [inline dkulib/dku_config/dss_parameter.py:48]:
```suggestion
        self.label = label or name
```
To make it more pythonic but if you think it is less readable, leave it that way, it works perfectly!

**Review by HenriChabert — CHANGES_REQUESTED** [review]
Very useful, thanks @Marsobad !

Can you bump the version in the __init__.py as well please?

**Comment by alexcombessie** [comment]
[ch62855](https://app.clubhouse.io/dataiku/story/62855/add-a-label-attribute-to-dssparameters-for-more-readable-error-messages)

**Comment by Marsobad** [comment]
> Very useful, thanks @Marsobad !
> 
> Can you bump the version in the **init**.py as well please?

I bumped the version in https://github.com/dataiku/dss-plugin-dkulib/pull/17/commits/ffaa0fc4a91aeb0ffb4d10413ed835bf4b5097ce

File: dkulib/dku_config/dss_parameter.py (hunk 2 above)
Comment (Marsobad) [reply dkulib/dku_config/dss_parameter.py]: done in https://github.com/dataiku/dss-plugin-dkulib/pull/17/commits/ffaa0fc4a91aeb0ffb4d10413ed835bf4b5097ce

File: dkulib/dku_config/dss_parameter.py (hunk 1 above)
Comment (Marsobad) [reply dkulib/dku_config/dss_parameter.py]: done in https://github.com/dataiku/dss-plugin-dkulib/pull/17/commits/ffaa0fc4a91aeb0ffb4d10413ed835bf4b5097ce

**Review by HenriChabert — APPROVED** [review]
LGTM

**Pull request merged**